### ISSUE TITLE
Consider resolving `~` or env vars in config values

### ISSUE THREAD

**Issue by ehaynes99** [flameshot-org/flameshot#3041]
### Feature Description

I share dotfiles across multiple machines, and they do not all have the same username. It would be really convenient if the config could be one of:
```
[General]
savePath=~/tmp/screenshots
```
```
[General]
savePath=$HOME/tmp/screenshots
```

Presumably, this would mean also allowing `%userprofile%`, etc. on windows

**Comment by mmahmoudian** [comment]
I agree that this is an essential thing for the same reason. We have discussed many times that the path expansion should not be done by Flameshot which I agree in general, but we need a way to also facilitate config migration and sync

**Comment by veracioux** [comment]
@mmahmoudian My view has changed since our previous discussions. This feature is sufficiently common, useful and not provided by other features, that I think we should implement it. But in doing this, we should probably add another setting `savePathExpandSpecialSymbols` (boolean).

I don't have any particular views about config migration and sync except my own use case:
I keep all my configs under git. It would be nice if we could have an additional config file that contains settings that change often, like `drawColor`, `drawFontSize`, `drawThickness`, etc. The reason is that `git status` shows them as changed, and that is annoying, at least to me.

**Comment by mmahmoudian** [comment]
@veracioux Why do we need to add a new expansion option? In my head if ~ is in the left-most position (e.g `~/`), it can be expanded, if there is anything before that (e.g `/~/`), it should be considered as literal. Although in %variable% and $variable it is tricky

About a local setting, I really like the idea👍. I also have my config in bare git repo and it hurts to see a dirty git status just because of some cache stuff. The screenkey and konsole are doing this annoying behavior as well. We should address that.
